### PR TITLE
Fix error in LZ4-compressed Zarr writing demo

### LINKED ISSUE #484: Double `else` in `benchmark_zarr_write_lz4_via_dask.py`
https://github.com/rapidsai/cucim/blob/8a660808c0a7b7513d3ce97cfd1743891877237c/examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py#L69-L75

This double `else` looks odd. What was intended?

Looking at the error message would have thought this

<details>

```suggestion
    if dtype == 'uint8':
        img = image_gpu
        assert img.dtype == cp.uint8
    elif dtype == 'uint16':
        img = image_gpu
        assert img.dtype == cp.uint16
    else:
        raise NotImplementedError("only testing for uint8 and uint16")
```

</details>

Though maybe the error is simply not needed? However then the `if` is probably unnecessary too

<details>

```suggestion
    img = image_gpu.astype(dtype)
```

</details>

_Originally posted by @jakirkham in https://github.com/rapidsai/cucim/pull/452#discussion_r1083065453_

cc @bdice @grlee77

### PR DESCRIPTION
**Pull request by grlee77**
closes #484

There was a mistake with double else statement. The intent was to restrict to uint8 or uint16 dtype as needed by the LZ4 compressor.

### CODE REVIEW

File: examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py [hunk 1]
```diff
@@ -66,13 +66,15 @@
 write_time_means = np.zeros(((len(dtypes), len(chunk_shapes), len(compressors), 2)), dtype=float)
 write_time_stds = np.zeros_like(write_time_means)
 for i, dtype in enumerate(dtypes):
```
Comment (jakirkham) [inline examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py:68]:
```suggestion
for i, dtype in enumerate(dtypes):
    dtype = np.dtype(dtype)
```

File: examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py [hunk 2]
```diff
@@ -66,13 +66,15 @@
...
-    if dtype == 'uint8':
+    if np.dtype(dtype) == np.uint8:
```
Comment (jakirkham) [inline examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py:69]:
```suggestion
    if dtype == np.uint8:
```

File: examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py [hunk 3]
```diff
@@ -66,13 +66,15 @@
...
         img = image_gpu
         assert img.dtype == cp.uint8
-    else:
+    elif np.dtype(dtype) == np.uint16:
```
Comment (jakirkham) [inline examples/python/gds_whole_slide/benchmark_zarr_write_lz4_via_dask.py:72]:
```suggestion
    elif dtype == np.uint16:
```

**Review by jakirkham — COMMENTED** [review]
Thanks Greg! 🙏

Would it make sense to do this conversion once up top?

**Comment by grlee77** [comment]
> Would it make sense to do this conversion once up top?

Yes, thanks John. I have applied the suggestions

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Greg! 🙏

**Pull request merged**